subsys/bluetooth/host/shell/bt_shell_private.c

```diff
@@ -0,0 +1,101 @@
+/**
+ * @file bt_shell_private.c
+ * @brief Bluetooth shell private module
+ *
+ * Provide common function which can be shared using privately inside Bluetooth shell.
+ */
+
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/shell/shell_backend.h>
+#include "bt_shell_private.h"
+
+void bt_shell_wall_vfprintf(enum shell_vt100_color color, const char *fmt, va_list args)
+{
+	int count;
+	const struct shell *sh;
+
+	count = shell_backend_count_get();
+	for (int i = 0; i < count; i++) {
+		va_list args_copy;
+
+		va_copy(args_copy, args);   /* Create a copy of 'args' to safely reuse */
+		sh = shell_backend_get(i);
+		shell_vfprintf(sh, color, fmt, args_copy);
+		va_end(args_copy);          /* Clean up to prevent resource leaks */
+	}
+}
+
+void bt_shell_hexdump(const uint8_t *data, size_t len)
+{
+	int count;
+	const struct shell *sh;
+
+	count = shell_backend_count_get();
+	for (int i = 0; i < count; i++) {
+		sh = shell_backend_get(i);
+		shell_hexdump(sh, data, len);
+	}
+}
+
+void bt_shell_help(void)
+{
+	int count;
+	const struct shell *sh;
+
+	count = shell_backend_count_get();
+	for (int i = 0; i < count; i++) {
+		sh = shell_backend_get(i);
+		shell_help(sh);
+	}
+}
```

Comment:
Since these 2 prints on all shells, should they have `wall` in the name too?

Reply:
Since the intention was to replace the `shell_*` functions that depended on `ctx_shell` with the appropriate `bt_shell_*` functions (in-short just adding `bt_` prefix) so I think it's no need to have `wall` in the name. :)

Reply:
That seems to be inconsistent with `bt_shell_wall_vfprintf` which servers the same purpose, but has `wall` in the name, right?

Reply:
|shell (variant)|bt_shell (variant) (before)|bt_shell (variant) (opt 1)|bt_shell (variant) (opt 2)|
|---|---|---|---|
|shell_fprintf   |bt_shell_fprintf   |bt_shell_fprintf   |bt_shell_**wall**_fprintf   |
|shell_fprintf_info|bt_shell_fprintf_info   |bt_shell_fprintf_info   |bt_shell_**wall**_fprintf_info   |
|shell_fprintf_print|bt_shell_fprintf_print   |bt_shell_fprintf_print   |bt_shell_**wall**_fprintf_print   |
|shell_fprintf_warn |bt_shell_fprintf_warn|bt_shell_fprintf_warn|bt_shell_**wall**_fprintf_warn|
|shell_fprintf_error|bt_shell_fprintf_error|bt_shell_fprintf_error|bt_shell_**wall**_fprintf_error|
|shell_hexdump|bt_shell_hexdump|bt_shell_**wall**_hexdump|bt_shell_**wall**_hexdump|
|shell_help|bt_shell_help|bt_shell_**wall**_help|bt_shell_**wall**_help|
|shell_info|bt_shell_info|bt_shell_info|bt_shell_**wall**_info|
|shell_print|bt_shell_print|bt_shell_print|bt_shell_**wall**_print|
|shell_warn|bt_shell_warn|bt_shell_warn|bt_shell_**wall**_warn|
|shell_error|bt_shell_error|bt_shell_error|bt_shell_**wall**_error|

---

Just bring out the table of print functions provided by `bt_shell_private.h`.
Which option are we going to choose: opt1, opt2, or something else?

Reply:
Since they are all `wall` in the end, then either the current, or opt 2. 

The main issue I had is that we had a single function using the `wall` name which seemed inconsistent as they are all `wall` variants. 

Of course since `bt_shell_wall_vfprintf` was an internal/static function not exposed by the header, it is less of any issue. 

So I'm OK with the current version or opt 2

Reply:
Thanks for the insight! I'll go with the current version. 🫡